File: src/brs/db/store/AssetTransferStore.java
package brs.db.store;

import brs.AssetTransfer;
import brs.db.SignumKey;
import brs.db.sql.EntitySqlTable;

import java.util.Collection;

public interface AssetTransferStore {
  SignumKey.LinkKeyFactory<AssetTransfer> getTransferDbKeyFactory();

  EntitySqlTable<AssetTransfer> getAssetTransferTable();

  Collection<AssetTransfer> getAssetTransfers(long assetId, int from, int to);

  Collection<AssetTransfer> getAccountAssetTransfers(long accountId, int from, int to);

  Collection<AssetTransfer> getAccountAssetTransfers(long accountId, long assetId, int from, int to);

  int getTransferCount(long assetId);
}

File: src/main/java/brs/db/store/AssetTransferStore.kt
package brs.db.store

import brs.AssetTransfer
import brs.db.BurstKey
import brs.db.sql.EntitySqlTable

interface AssetTransferStore {
    val transferDbKeyFactory: BurstKey.LongKeyFactory<AssetTransfer>

    val assetTransferTable: EntitySqlTable<AssetTransfer>

    suspend fun getAssetTransfers(assetId: Long, from: Int, to: Int): Collection<AssetTransfer>

    suspend fun getAccountAssetTransfers(accountId: Long, from: Int, to: Int): Collection<AssetTransfer>

    suspend fun getAccountAssetTransfers(accountId: Long, assetId: Long, from: Int, to: Int): Collection<AssetTransfer>

    suspend fun getTransferCount(assetId: Long): Int
}
